osdep/timer-linux.c

```diff
@@ -35,11 +35,12 @@ void mp_sleep_ns(int64_t ns)
 uint64_t mp_raw_time_ns(void)
 {
     struct timespec tp = {0};
+    int ret = -1;
 #if defined(CLOCK_MONOTONIC_RAW)
-    clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
-#else
-    timespec_get(&tp, TIME_UTC);
+    ret = clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
 #endif
+    if (ret)
+        clock_gettime(CLOCK_MONOTONIC, &tp);
```

Comment:
Note that there's a small discrepancy between linux and BSDs with `CLOCK_MONOTONIC`. Linux doesn't follow POSIX and doesn't advance the clock during system suspension (and fixing it [broke some user-land applications](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a3ed0e4393d6885b4af7ce84b437dc696490a530) so it remains "broken").

Does this matter for mpv? I don't know, which is why I'm pointing it out here. If it does, then on linux you'd need to use `CLOCK_BOOTTIME` instead.

Reply:
We were using `CLOCK_MONOTONIC` before for many years anyway on both BSD and Linux. In any case, linux should be using `CLOCK_MONOTONIC_RAW` anyways.

Reply:
> linux should be using `CLOCK_MONOTONIC_RAW` anyways.

That also doesn't advance when system is suspended.

https://www.man7.org/linux/man-pages/man3/clock_gettime.3.html
>This clock does not count time that the system is suspended.

Reply:
Well still it shouldn't matter seeing as this was `CLOCK_MONOTONIC` for like 10 years.